pkg/kubelet/rkt/pod_info.go

```diff
@@ -0,0 +1,225 @@
+			// TODO(yifan): Parse the status here. This requires more details in
+			// the rkt status, (e.g. started time, image name, etc).
+			p.state = tuples[1]
+		case "networks":
+			p.ip = getIPFromNetworkInfo(tuples[1])
+		case "pid":
+			pid, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				glog.Errorf("Cannot parse pid %q: %v", tuples[1], err)
+				continue
+			}
+			p.pid = pid
+		}
+		if strings.HasPrefix(tuples[0], "sha512") {
+			exitcode, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				glog.Errorf("Cannot parse exit code %q: %v", tuples[1], err)
+			} else {
+				p.exitCodes[tuples[0]] = exitcode
+			}
+		}
+	}
+	return nil
+}
+
+// getIPFromNetworkInfo returns the IP of a pod by parsing the network info.
+// The network info looks like this:
+//
+// default:ip4=172.16.28.3, database:ip4=172.16.28.42
+//
+func getIPFromNetworkInfo(networkInfo string) string {
+	parts := strings.Split(networkInfo, ",")
+
+	for _, part := range parts {
+		if strings.HasPrefix(part, "default:") {
+			tuples := strings.Split(part, "=")
+			if len(tuples) == 2 {
+				return tuples[1]
+			}
+		}
+	}
+	return ""
+}
+
+// getContainerStatus creates the api.containerStatus of a container from the podInfo.
+// TODO(yifan): Get more detailed info such as Image, ImageID, etc.
+func (p *podInfo) getContainerStatus(container *kubecontainer.Container) api.ContainerStatus {
+	var status api.ContainerStatus
+	status.Name = container.Name
+	status.Image = container.Image
+
+	containerID, _ := parseContainerID(string(container.ID))
+	status.ImageID = containerID.imageID
+
+	switch p.state {
+	case Running:
+		// TODO(yifan): Get StartedAt.
+		status.State = api.ContainerState{
+			Running: &api.ContainerStateRunning{
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	case Embryo, Preparing, Prepared:
+		status.State = api.ContainerState{Waiting: &api.ContainerStateWaiting{}}
+	case AbortedPrepare, Deleting, Exited, Garbage:
+		exitCode, ok := p.exitCodes[status.ImageID]
+		if !ok {
+			glog.Warningf("rkt: Cannot get exit code for container %v", container)
+		}
+		exitCode = -1
+		status.State = api.ContainerState{
+			Termination: &api.ContainerStateTerminated{
+				ExitCode:  exitCode,
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	default:
+		glog.Warningf("rkt: Unknown pod state: %q", p.state)
+	}
+	return status
+}
+
+// toPodStatus converts a podInfo type into an api.PodStatus type.
+func (p *podInfo) toPodStatus(pod *kubecontainer.Pod) api.PodStatus {
+	var status api.PodStatus
+	status.PodIP = p.ip
+	// For now just make every container's state the same as the pod.
+	for _, container := range pod.Containers {
+		status.ContainerStatuses = append(status.ContainerStatuses, p.getContainerStatus(container))
+	}
+	return status
+}
+
+// splitLineByTab breaks a line by tabs, and trims the leading and tailing spaces.
+// Not using strings.Split because there could be multiple tabs.
+func splitLineByTab(line string) []string {
```

Comment:
Alternatively: Split on tabs and remove all empty strings in the slice. May be simpler to implement and reason about.